### ISSUE TITLE
Call to a member function bindValue() on a non-object

### ISSUE THREAD

**Issue by KataiKi** [panique/php-login-one-file#21]
When registering a new user, I get this error.
Fatal error: Call to a member function bindValue() on a non-object in C:\inetpub\wwwroot\index.php on line 308

Running PHP 5.5, so I'm not sure what's going on.  Logging in gives the same error on line 212

**Comment by MikeBlane** [comment]
Panique, I am using a hosted Linux CentOS system and they have PHP 5.4 installed.  I am receiving the same message.  I have googled the error message and am not seeing the solution to this. I should also explain that I have a rudimentary understanding and only very simple practical experience of object oriented programming.  
I would really appreciate any help you could offer.

**Comment by panique** [comment]
@MikeBlane https://github.com/panique/php-login-one-file#requirements you probably have not the minimum requirements or you have changed the code maybe ?

**Comment by MikeBlane** [comment]
The requirements are met.  The server I am using has PHP 5.4 installed.
When I execute the "_install.php" script, I am given the following message:

Database ../users.db was created, installation was successful.

I do not remember changing the index.php file, but I re-downloaded and copied to the server to ensure that I did not change anything.  With the newly created database, I try to login before setting up an account and receive this (using username "user" and password "password", which I have not created yet):

Fatal error: Call to a member function bindValue() on a non-object in /home1/x101deg/public_html/dispatch/index.php on line 212

If I try to create the user "user" with password "password", I get:

Fatal error: Call to a member function bindValue() on a non-object in /home1/x101deg/public_html/dispatch/index.php on line 212

When I run _debug.php, I get:

array(0) { }

**Comment by panique** [comment]
Sorry, I cannot offer free personal help. If you have problems with this and are not able to fix this by yourself, then tihs script might not be for you. Sorry.

**Comment by jackmawer** [comment]
I'm still having this issue, and the support forum is closed!

**Comment by panique** [comment]
@jackmawer Okay, and now ? You have not even given a minimum of useful information about your problem. Guys, learn to use the internet! You want help ? Then give something others can work with. "I have the same problem" is not useful.

**Comment by jackmawer** [comment]
@panique Actually, I

**Comment by jackmawer** [comment]
@panique And now you attempt to replicate the bug, kindly asking for information where required. I know how to use the internet, thanks, and you have plenty to work with from the above. I am telling you that I can replicate the issue, therefore, I *am* giving you useful information! Now stop being so snappy and GROW UP.

**Comment by panique** [comment]
@jackmawer Seriously man, I spend a lot of time answering support requests of random strangers, for FREE. And people who are not clever enough to read a readme, follow a install guideline and who expect detailed help for "questions" that are like "i does not work".

I mean, seriously, what do you want ? Follow the install guidelines and everything is fine. if you want help, show your code, show the exact problem, show what you have tried etc.

Sometimes I'm asking myself why I'm even answering all this stuff.

**Comment by Stuart98** [comment]
I was also having the same problem, so thought I would share my solution to help anyone else hitting this thread in the future.

Basically the issue was the path to my "users.db" file was incorrect which is set in the index.php. I had the index.php file in a subdirectory and included from a root level file, something like:

includes
....login
........index.php
........users.db
myAppFile.php

I updated the path in the index.php to reflect this:

    private $db_sqlite_path = "_includes/login/users.db";

which seems to have solved the problem.

For some more debugging info I used the PDO `errorInfo` function (`print_r($this->db_connection->errorInfo())`) to realise that the SQL query was being rejected because it couldn't find the table 'users'. This was because it was creating a new users.db file when connecting (not the one created after running the install script) which was empty.

I hope this helps someone else.
